modules/javafx.base/src/main/java/com/sun/javafx/collections/ObservableSequentialListWrapper.java

```diff
@@ -175,6 +175,14 @@ public E get(int index) {
 
     @Override
     public boolean addAll(int index, Collection<? extends E> c) {
+        if (c.isEmpty()) {
+            if (index < 0 || index > size()) {
+                throw new IndexOutOfBoundsException("Index: " + index);
+            }
+
+            return false;
+        }
+
```

Comment:
Shouldn't this always check the index?

```suggestion
        if (index < 0 || index > size()) {
            throw new IndexOutOfBoundsException("Index: " + index);
        }
        if (c.isEmpty()) {   // implicit null check
            return false;
        }
```